src/util/output/builds.js

```diff
@@ -62,49 +141,196 @@ const styleOutput = (output) => {
+    return partCounter > prev ? partCounter : prev;
+  }, 0);
+
+  // Group builds together
+  let counter = 0;
+
+  while (path.length > MAX_BUILD_GROUPS) {
+    path = groupBuilds(path, highestLevel, counter);
+    counter++;
+  }
+
+  // Reverse the paths so that the larger chunks come last
+  // including the / directory
+  path = path.reverse();
+
   const longestSource = builds.reduce((final, current) => {
     const { length } = current.entrypoint;
     return length > final ? length : final;
   }, 0);
 
-  const final = buildsAndOutput.map((item, index) => {
-    let log = null;
+  const final = [];
+  let finalBuildsLength = path.length;
+  let lengthWithoutRootPaths = path.length;
+  let hiddenBuildGroup = [];
+
+  // Ungroup the root files
+  path = (() => {
+    const nextList = [];
+    const rootList = [];
+
+    for (const group of path) {
+      if (getCommonPath(group) === '/') {
+        group.map((item) => rootList.push([item]));
+      } else {
+        nextList.push(group);
+      }
+    }
+
+    lengthWithoutRootPaths = nextList.length;
+    rootList.map((group) => nextList.push(group));
+
+    return nextList;
+  })();
+
+  path.map((buildGroup) => {
+    const commonPath = getCommonPath(buildGroup);
 
-    if (item.isOutput) {
-      log = styleOutput(item);
+    // All items with the common path / are a single group
+    if (commonPath === '/') {
+      if (
+        lengthWithoutRootPaths <= MAX_BUILD_GROUPS &&
+        finalBuildsLength <= MAX_BUILD_GROUPS
+      ) {
+        const item = buildGroup[0];
+        final.push(`${styleBuild(item, times, longestSource)}\n`);
+        finalBuildsLength++;
+      } else {
+        hiddenBuildGroup.push(buildGroup[0]);
+        return buildGroup;
+      }
+    } else if (buildGroup.length === 1) {
+      const item = buildGroup[0];
+      final.push(`${styleBuild(item, times, longestSource)}\n`);
+      finalBuildsLength++;
     } else {
-      log = styleBuild(item, times, longestSource);
+      final.push(`${styleHiddenBuilds(`${commonPath}/*`, buildGroup, times, longestSource)}\n`);
+      finalBuildsLength++;
+    }
+
+    // Get the first five outputs when the deployment is ready
+    const outputs = buildGroup.reduce((prevValue, currentValue) => (
+      prevValue.concat(Array.isArray(currentValue.output)
+        ? currentValue.output
+        : []
+      )
+    ), []);
+
+    outputs.slice(0, MAX_OUTPUTS_PER_GROUP).map((output, index) => (
+      final.push(`${styleOutput({
+        ...output,
+        isLast: outputs.length === (index + 1)
+      })}\n`)
+    ));
+
+    if (outputs.length > MAX_OUTPUTS_PER_GROUP) {
+      final.push(chalk.grey(`└── ${outputs.length - MAX_OUTPUTS_PER_GROUP} hidden\n`));
     }
 
-    const newline = index === buildsAndOutput.length - 1 ? '' : '\n';
-    return log + newline;
+    return buildGroup;
   });
 
+  if (hiddenBuildGroup.length) {
+    final.push(`${styleHiddenBuilds(
+      `${hiddenBuildGroup.length} hidden`,
```

Comment:
For outputs, let's show `x output items hidden`, for builds show `x builds hidden`